das/google_cloud_storage_service.go

```diff
@@ -0,0 +1,158 @@
+package das
+
+import (
+	googlestorage "cloud.google.com/go/storage"
+	"context"
+	"encoding/hex"
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/google/go-cmp/cmp"
+	"github.com/offchainlabs/nitro/arbstate/daprovider"
+	"github.com/offchainlabs/nitro/das/dastree"
+	"github.com/offchainlabs/nitro/util/pretty"
+	flag "github.com/spf13/pflag"
+	"google.golang.org/api/option"
+	"io"
+	"sort"
+)
+
+type GoogleCloudStorageOperator interface {
+	Bucket(name string) *googlestorage.BucketHandle
+	Upload(ctx context.Context, bucket, objectPrefix string, value []byte) error
+	Download(ctx context.Context, bucket, objectPrefix string, key common.Hash) ([]byte, error)
+	Close(ctx context.Context) error
+}
+
+type GoogleCloudStorageClient struct {
+	client *googlestorage.Client
+}
+
+func (g *GoogleCloudStorageClient) Bucket(name string) *googlestorage.BucketHandle {
+	return g.client.Bucket(name)
+}
+
+func (g *GoogleCloudStorageClient) Upload(ctx context.Context, bucket, objectPrefix string, value []byte) error {
+	obj := g.client.Bucket(bucket).Object(objectPrefix + EncodeStorageServiceKey(dastree.Hash(value)))
+	w := obj.NewWriter(ctx)
+
+	if _, err := fmt.Fprintln(w, hex.EncodeToString(value)); err != nil {
```

Comment:
I'm not a GCS expert, is it necessary to hex encode the data here? Otherwise you're just doubling the size for no reason.

Reply:
I was thinking if it is hex, would be much easier to do preview, but seems like both s3 and google cloud storage doesn't allow you to preview on website, you need download it. So i am following your suggestion to just store raw bytes.